tools/sil-opt/SILOpt.cpp

```diff
@@ -296,6 +296,18 @@ static llvm::cl::opt<OptimizationMode> OptModeFlag(
                                 "ignore debug info, reduce runtime")),
     llvm::cl::init(OptimizationMode::NotSet));
 
+static llvm::cl::opt<IRGenDebugInfoLevel> IRGenDebugInfoLevelArg(
+    "irgen-debuginfo-level", llvm::cl::desc("IRGen debug info level"),
+    llvm::cl::values(clEnumValN(IRGenDebugInfoLevel::None, "none",
+                                "No debug info"),
+                     clEnumValN(IRGenDebugInfoLevel::LineTables, "line-tables",
+                                "Line tables only"),
+                     clEnumValN(IRGenDebugInfoLevel::ASTTypes, "ast-types",
+                                "Line tables + AST type references"),
+                     clEnumValN(IRGenDebugInfoLevel::DwarfTypes, "dwarf-types",
+                                "Line tables + AST type refs + Dwarf types")),
+    llvm::cl::init(IRGenDebugInfoLevel::ASTTypes));
```

Comment:
What is the `IRGenDebugInfoLevelArg` actually supposed to do and why does it default to ASTTypes? It doesn't appear to be based on the current debug mode. Does it essentially strip any SIL debug information that may happen to have been emitted for the current debug mode if that information isn't required by the selected IRGen debug mode?